layer1/layer1.yaml

```diff
@@ -65,9 +67,21 @@ components:
         flow_control:
           $ref: '#/components/schemas/Layer1.FlowControl'
     
+    Layer1.EthernetVirtual:
+      description: >-
+        Container for virtual ethernet settings
+      type: object
+      properties:
+        speed:
+          description: >-
+            This is the speed that will be set on the virtual ethernet interface.
+          type: string
+          enum: [one_hundred_mbps, one_gbps, ten_gbps, twenty_five_gpbs, fity_gbps, one_hundred_gpbs]
```

Comment:
can we do 100mbps, 1gbps, 10gbps, 25gbps, 50gbps, 100gbps (btw, just noticed typo on fity_gbps and one_hundred_gpbs)?

Reply:
We should switch all enums that have numbers to that model then.

Reply:
and make it part of the modeling guide

Reply:
Please note, enum names starting with numbers are not very portable across languages (i mean sure this would pass code generation, but it may transform the names to something unwanted).

Reply:
Do you have examples?  Which languages?  If it is breaking in some language(s) that we don't care about then, we could ignore it.

Reply:
Summary: bad for *.proto.

Works fine with Python (since enum in python is a string which is validated on client side).
And Go too for similar reason (except enum validation doesn't work so they're reconsidering other approach: https://github.com/go-swagger/go-swagger/issues/861)

On protobuf, this is the generated code (note leading underscore). When code is generated from this, the field names will be further transformed into something without leading underscore (https://github.com/golang/protobuf/issues/201).
```protobuf

  // This is the speed that will be used if auto_negotiate is false.
  enum speed {
    _1000_MBPS = 0;
    _100_FD_MBPS = 1;
    _100_HD_MBPS = 2;
    _10_FD_MBPS = 3;
    _10_HD_MBPS = 4;
  }
```

Reply:
This should be reverted to the original.